src/Database/LSMTree/Internal/Snapshot.hs

```diff
@@ -111,12 +120,31 @@ data SnapshotMetaData = SnapshotMetaData {
     -- | The write buffer.
   , snapWriteBuffer   :: !RunNumber
     -- | The shape of the levels of the LSM tree.
-  , snapMetaLevels    :: !(SnapLevels RunNumber)
+  , snapMetaLevels    :: !(SnapLevels SnapshotRun)
+    -- | The state of tree merging of the LSM tree.
+  , snapMergingTree   :: !(Maybe (SnapMergingTree SnapshotRun))
   }
   deriving stock Eq
 
 instance NFData SnapshotMetaData where
-  rnf (SnapshotMetaData a b c d e) = rnf a `seq` rnf b `seq` rnf c `seq` rnf d `seq` rnf e
+  rnf (SnapshotMetaData a b c d e f) =
+    rnf a `seq` rnf b `seq` rnf c `seq`
+    rnf d `seq` rnf e `seq` rnf f
+
+-- | Information needed to open a 'Run' from disk, see 'Run.openFromDisk'.
+--
+-- TODO: one could imagine needing only the 'RunNumber' to identify the files
+-- on disk, and the other parameters being stored with the run itself, rather
+-- than needing to be supplied.
+data SnapshotRun = SnapshotRun {
+       snapRunNumber  :: !RunNumber,
+       snapRunCaching :: !Run.RunDataCaching,
+       snapRunIndex   :: !Run.IndexType
+     }
+  deriving stock Eq
+
+instance NFData SnapshotRun where
+  rnf (SnapshotRun a b c) = rnf a `seq` rnf b `seq` rnf c
```

Comment:
Maybe put this in the `Runs` section

Reply:
I'll clarify that that this is the type at which we instantiate both `SnapLevels` and `SnapMergingTree` in the `SnapshotMetaData`. And that it's generally used as an intermediate format between the serialised version and the semi-rehydrated version where we traverse to open all the runs.

Reply:
Actually I've changed my mind. I agree with your suggestion.